.github/workflows/benchmark-flow.yml

```diff
@@ -0,0 +1,118 @@
+        type: number
+        default: 1
+      benchmark_runner_group_total:
+        type: number
+        default: 1
+
+jobs:
+  benchmark-steps:
+    runs-on: ubuntu-latest
+    container: ${{ inputs.container }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - run: |
+          git init
+          git config --global --add safe.directory '*'
+          git submodule update --init --recursive
+
+      - name: System setup
+        run: ./sbin/system-setup.py
+
+      - name: Get Redis
+        uses: actions/checkout@v4
+        with:
+          repository: redis/redis
+          ref: '7.2'
+          path: redis
+
+      - name: Build Redis
+        working-directory: redis
+        run: make install
+
+      - name: Build RedisJSON
+        run: |
+          . $HOME/.cargo/env
+          make build
+
+      - name: Prepare automation
+        run: |
+          ./deps/readies/bin/getpy3
+          python3 -m pip install -r tests/benchmarks/requirements.txt
+          VERSION=0.14.8 ./deps/readies/bin/getterraform
+
+      - name: Run CI benchmarks on aws for envs ${{ inputs.allowed_envs }}
+        timeout-minutes: 240 # timeout for the step
+        working-directory: tests/benchmarks
+        continue-on-error: true
+        env:
+          # Hard-coded
+          PERF_CALLGRAPH_MODE: dwarf
+          MAX_PROFILERS: 1
+          # Secrets
+          AWS_ACCESS_KEY_ID: ${{ secrets.PERFORMANCE_EC2_ACCESS_KEY }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.PERFORMANCE_EC2_SECRET_KEY }}
+          AWS_DEFAULT_REGION: ${{ secrets.PERFORMANCE_EC2_REGION }}
+          EC2_PRIVATE_PEM: ${{ secrets.PERFORMANCE_EC2_PRIVATE_PEM }}
+          # Inputs
+          PROFILE: ${{ inputs.profile_env }}
+          BENCHMARK_GLOB: ${{ inputs.benchmark_glob }}
+          BENCHMARK_RUNNER_GROUP_M_ID: ${{ inputs.benchmark_runner_group_member_id }}
+          BENCHMARK_RUNNER_GROUP_TOTAL: ${{ inputs.benchmark_runner_group_total }}
+        run: redisbench-admin run-remote
+              --module_path ../../${{ inputs.module_path }}
+              --github_actor ${{ github.triggering_actor }}
+              --github_repo ${{ github.event.repository.name }}
+              --github_org ${{ github.repository_owner }}
+              --required-module ReJSON
+              --github_sha ${{ github.sha }}
+              --github_branch ${{ github.head_ref || github.ref_name }}
+              --upload_results_s3
+              --triggering_env ${{ inputs.triggering_env }}
+              --allowed-envs ${{ inputs.allowed_envs }}
+              --allowed-setups ${{ inputs.allowed_setups }}
+              --push_results_redistimeseries
+              --redistimeseries_host ${{ secrets.PERFORMANCE_RTS_HOST }}
+              --redistimeseries_port ${{ secrets.PERFORMANCE_RTS_PORT }}
+              --redistimeseries_pass '${{ secrets.PERFORMANCE_RTS_AUTH }}'
+
+      - name: Generate Pull Request Performance info
+        if: github.event.number
```

Comment:
Worth mentioning that I wasn’t able to verify that this will work as expected when the trigger is not a pull request. We should verify it when it is merged, or we can try and test it on another dummy repo

Reply:
@GuyAv46 indeed. Lets check after merge